### PR TITLE
feat: add tracing to the NATS message bus

### PR DESCRIPTION
**Pull request by tiagolobocastro**
This means we can now see the trace stack from a compatible rest client
all the way to the core agent.

### CODE REVIEW

File: common/src/mbus_api/send.rs [hunk 1]
```diff
@@ -296,17 +317,69 @@ where
...
+                opentelemetry_semantic_conventions::trace::MESSAGING_DESTINATION,
+                self.channel.to_string(),
+            ),
+            KeyValue::new(
+                opentelemetry_semantic_conventions::trace::MESSAGING_MESSAGE_ID,
+                self.payload.id.to_string(),
+            ),
+        ];
+
+        let cx = Context::current();
```
Comment (paulyoong) [inline common/src/mbus_api/send.rs:337]: Nit: cx -> ctx?

**Review by paulyoong — APPROVED** [review]
Awesome!

File: common/src/mbus_api/send.rs (hunk 1 above)
Comment (tiagolobocastro) [reply common/src/mbus_api/send.rs:337]: oh yeah indeed

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**